llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -15887,29 +15887,29 @@ const SCEV *ScalarEvolution::LoopGuards::rewrite(const SCEV *Expr) const {
         return SCEVRewriteVisitor<SCEVLoopGuardRewriter>::visitZeroExtendExpr(
             Expr);
       }
-      return I->second;
+      return S;
     }
 
     const SCEV *visitSignExtendExpr(const SCEVSignExtendExpr *Expr) {
-      auto I = Map.find(Expr);
-      if (I == Map.end())
+      const SCEV *S = Map.lookup(Expr);
+      if (!S)
```

Comment:
nit: with `lookup`, we could also fold it into the `if`, limiting the scope, like

```suggestion
      if (const SCEV *S = Map.lookup(Expr))
        return S;
```